File: src/main/java/work/lclpnet/ap2/game/cozy_campfire/setup/CCBase.java
package work.lclpnet.ap2.game.cozy_campfire.setup;

import net.minecraft.entity.Entity;
import net.minecraft.util.math.BlockPos;
import work.lclpnet.ap2.api.util.Collider;
import work.lclpnet.ap2.impl.util.BlockBox;
import work.lclpnet.ap2.impl.util.collision.UnionCollider;

import java.util.List;
import java.util.UUID;

public class CCBase {

    private final Collider bounds;
    private final BlockPos campfirePos;
    private final UUID entityUuid;

    public CCBase(List<BlockBox> bounds, BlockPos campfirePos, UUID entityUuid) {
        this.bounds = new UnionCollider(bounds.toArray(BlockBox[]::new));
        this.campfirePos = campfirePos;
        this.entityUuid = entityUuid;
    }

    public boolean isEntity(Entity entity) {
        return entityUuid.equals(entity.getUuid());
    }

    public BlockPos getCampfirePos() {
        return campfirePos;
    }

    public Collider getBounds() {
        return bounds;
    }

    public boolean isInside(double x, double y, double z) {
        return bounds.collidesWith(x, y, z);
    }
}

File: src/minigames/cozy_campfire/src/main/kotlin/work/lclpnet/ap2/game/cozy_campfire/setup/CCBase.kt
package work.lclpnet.ap2.game.cozy_campfire.setup

import net.minecraft.core.BlockPos
import net.minecraft.world.entity.Entity
import work.lclpnet.gaco.collisions.UnionCollider
import work.lclpnet.gaco.ds.BlockBox
import work.lclpnet.gaco.ds.Collider
import work.lclpnet.kibu.structure.BlockStructure
import java.util.UUID

data class CCBase(
    val bounds: Collider,
    val campfirePos: BlockPos,
    val entityUuid: UUID,
    val doorSchematic: BlockStructure?,
    val doorPos: BlockPos?
) {
    constructor(
        bounds: List<BlockBox>,
        campfirePos: BlockPos,
        entityUuid: UUID,
        doorSchematic: BlockStructure?,
        doorPos: BlockPos?
    ) : this(
        UnionCollider(bounds.toTypedArray()),
        campfirePos,
        entityUuid,
        doorSchematic,
        doorPos
    )

    fun isEntity(entity: Entity) = entityUuid == entity.uuid
    fun isInside(x: Double, y: Double, z: Double) = bounds.collidesWith(x, y, z)
}
